src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -244,14 +246,71 @@ export class DefaultTelemetryService implements TelemetryService {
 
     private static readEventsFromCache(cachePath: string): TelemetryEvent[] {
         try {
-            const events = JSON.parse(fs.readFileSync(cachePath, 'utf-8')) as TelemetryEvent[]
+            const input = JSON.parse(fs.readFileSync(cachePath, 'utf-8'))
+            const events = filterTelemetryCacheEvents(input)
             events.forEach((element: TelemetryEvent) => {
                 element.createTime = new Date(element.createTime)
             })
 
             return events
-        } catch {
+        } catch (error) {
+            // tslint:disable-next-line: no-unsafe-any
+            getLogger().error(error)
+
             return []
         }
     }
 }
+
+export function filterTelemetryCacheEvents(input: any): TelemetryEvent[] {
+    if (!Array.isArray(input)) {
+        getLogger().error(`Input into filterTelemetryCacheEvents:\n${input}\nis not an array!`)
+
+        return []
+    }
+    const arr = input as any[]
+
+    return arr
+        .filter((item: any) => {
+            // Make sure the item is an object
+            if (item !== Object(item)) {
+                getLogger().error(`Item in telemetry cache:\n${item}\nis not an object! skipping!`)
+
+                return false
+            }
+
+            return true
+        })
+        .filter((item: Object) => {
+            // Only accept objects that have createTime and data because that's what's required by TelemetryEvent
+            if (!item.hasOwnProperty('createTime') || !item.hasOwnProperty('data')) {
+                getLogger().warn(`Item in telemetry cache: ${item}\n does not have 'data' or 'createTime'! skipping!`)
```

Comment:
Going back to side question above: does `createTime` matter if we're immediately overwriting it?